supervisor/api/addons.py

```diff
@@ -315,6 +317,11 @@ async def security(self, request: web.Request) -> None:
         addon: AnyAddon = self._extract_addon(request)
         body: Dict[str, Any] = await api_validate(SCHEMA_SECURITY, request)
 
+        if not isinstance(addon, Addon):
+            raise APIError(
+                f"Cannot set security options, add-on {addon.slug} is not installed"
+            )
```

Comment:
That is handle by _extract_addon

Reply:
Yes, agreed, however, when type hinting, the `_extract_addon` returns `AnyAddon`, which can be an instance of `Addon` or `AddonStore`.

Since these methods (where I've added this check), rely on methods only exposed in `Addon` and are not available in `AddonStore`/`AddonModel`.

If this check is not here, it fails type checking.

I agree the `_extract_addon` is helpful, however, its return type is ambiguous.

Reply:
Which raises the error before this code happens...

Reply:
This is about type checking and hinting, not runtime.
Sure, at runtime it is captured, from a type checking perspective, it isn't, as `_extract_addon` returns `AnyAddon`, it means that any static type checking tool will take into account it may be `AddonStore` as well.

Reply:
Let's split _extract_addon into two functions to make mypi happy. I don't like to maintain code which never can happen.

Reply:
Done